cmd/tailscale/cli/serve.go

```diff
@@ -333,12 +331,31 @@ func (e *serveEnv) runServe(ctx context.Context, args []string) error {
 	return nil
 }
 
-func (e *serveEnv) handleWebServeRemove(ctx context.Context, mount string) error {
-	srvPort, err := e.validateServePort()
-	if err != nil {
-		return err
+func isProxyTarget(source string) bool {
+	if strings.HasPrefix(source, "http") && strings.Index(source, "://") > 0 {
+		return true
+	}
+	// support "localhost:3000", for example
+	_, portStr, _ := strings.Cut(source, ":")
+	if allNumeric(portStr) && strings.HasSuffix(source, ":"+portStr) {
```

Comment:
`strings.HasSuffix(source, ":"+portStr)` is going to be always true based on `allNumeric(portStr)`

maybe
```
_, portStr, ok := strings.Cut(source, ":")
if ok && allNumeric(portStr) {
```

Reply:
✅